tests/run-tests.py

```diff
@@ -824,6 +824,12 @@ def run_one_test(test_file):
         output_mupy = run_micropython(pyb, args, test_file, test_file_abspath)
 
         if output_mupy == b"SKIP\n":
+            if pyb is not None and hasattr(pyb, "read_until"):
+                # Running on a target over a serial connection, and the target requested
+                # to skip the test.  It does this via a SystemExit which triggers a soft
+                # reset.  Wait for the soft reset to finish, so we don't interrupt the
+                # start-up code (eg boot.py) when preparing to run the next test.
+                pyb.read_until(1, b"raw REPL; CTRL-B to exit\r\n", timeout=1)
```

Comment:
I'm not sure if 1 second is really long enough here? 

I'm thinking of the likes of a wifi board that's used for multitests where it's common to have wifi & network setup in boot.py, this can take longer. 

Considering this will continue sooner when that startup message is received I'd be inclined to set a reasonably long timeout for the less common cases that need it?

Reply:
This is not the total timeout, but actually the timeout between characters received over the serial connection.

Well, you're right that it could still take a while, with nothing printed for many seconds.

I can increase it.  All boards are expected to do a soft reset after skipping a test, so it should only time out if something is wrong.

Reply:
I've now changed this to use the default timeout of 10 seconds.

Tested on stm32, rp2 and esp32 boards and the wait for the soft reset after a skip is quick (because they don't do much in their boot.py).

Reply:
That seems reasonable, looks good!